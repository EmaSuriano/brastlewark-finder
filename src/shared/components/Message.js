import styled from 'styled-components';

const Message = styled.p`
  margin: auto;
`;

export default Message;
