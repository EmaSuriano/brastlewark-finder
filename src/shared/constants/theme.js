import PropTypes from 'prop-types';

export const ThemePropTypes = PropTypes.shape({
  background: PropTypes.string,
  light: PropTypes.string,
  disabled: PropTypes.string,
  accent: PropTypes.string,
  secondary: PropTypes.string,
}).isRequired;

export default {
  background: '#4F5D75',
  light: '#FFFFFF',
  disabled: '#BFC0C0',
  accent: '#EF8354',
  secondary: '#2D3142',
};
