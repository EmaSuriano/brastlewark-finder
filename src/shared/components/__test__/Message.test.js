import React from 'react';
import { shallow } from 'enzyme';
import theme from 'shared/constants/theme';
import Message from '../Message';

describe('<Message />', () => {
  describe('render', () => {
    it('default props', () => {
      expect(shallow(<Message theme={theme} />)).toMatchSnapshot();
    });

    it('with color', () => {
      expect(
        shallow(<Message color="accent" theme={theme} />),
      ).toMatchSnapshot();
    });

    it('with bold', () => {
      expect(shallow(<Message bold theme={theme} />)).toMatchSnapshot();
    });

    it('with fontSize', () => {
      expect(
        shallow(<Message fontSize="12em" theme={theme} />),
      ).toMatchSnapshot();
    });
  });
});
