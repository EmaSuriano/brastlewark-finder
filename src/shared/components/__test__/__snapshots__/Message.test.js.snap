// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> render default props 1`] = `
.c0 {
  color: #FFFFFF;
  font-weight: inherit;
  font-size: inherit;
}

<p
  className="c0"
  color="light"
  fontSize="inherit"
/>
`;

exports[`<Message /> render with bold 1`] = `
.c0 {
  color: #FFFFFF;
  font-weight: bold;
  font-size: inherit;
}

<p
  className="c0"
  color="light"
  fontSize="inherit"
/>
`;

exports[`<Message /> render with color 1`] = `
.c0 {
  color: #EF8354;
  font-weight: inherit;
  font-size: inherit;
}

<p
  className="c0"
  color="accent"
  fontSize="inherit"
/>
`;

exports[`<Message /> render with fontSize 1`] = `
.c0 {
  color: #FFFFFF;
  font-weight: inherit;
  font-size: 12em;
}

<p
  className="c0"
  color="light"
  fontSize="12em"
/>
`;
